resource-management/pkg/common-lib/metrics/event_metrics.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2022 Authors of Global Resource Service.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"fmt"
+	"k8s.io/klog/v2"
+	"sync"
+
+	common_lib "global-resource-service/resource-management/pkg/common-lib"
+	"global-resource-service/resource-management/pkg/common-lib/types/runtime"
+)
+
+type LatencyMetricsAllCheckpoints struct {
+	Aggregator_Received  *LatencyMetrics
+	Distributor_Received *LatencyMetrics
+	Distributor_Sending  *LatencyMetrics
+	Distributor_Sent     *LatencyMetrics
+	Serializer_Encoded   *LatencyMetrics
+	Serializer_Sent      *LatencyMetrics
+}
+
+var latencyMetricsAllCheckpoints *LatencyMetricsAllCheckpoints
+var latencyMetricsLock sync.RWMutex
+
+func init() {
+	latencyMetricsAllCheckpoints = new(LatencyMetricsAllCheckpoints)
+	latencyMetricsAllCheckpoints.Aggregator_Received = NewLatencyMetrics(int(Aggregator_Received))
+	latencyMetricsAllCheckpoints.Distributor_Received = NewLatencyMetrics(int(Distributor_Received))
+	latencyMetricsAllCheckpoints.Distributor_Sending = NewLatencyMetrics(int(Distributor_Sending))
+	latencyMetricsAllCheckpoints.Distributor_Sent = NewLatencyMetrics(int(Distributor_Sent))
+	latencyMetricsAllCheckpoints.Serializer_Encoded = NewLatencyMetrics(int(Serializer_Encoded))
+	latencyMetricsAllCheckpoints.Serializer_Sent = NewLatencyMetrics(int(Serializer_Sent))
+}
+
+func AddLatencyMetricsAllCheckpoints(e runtime.Object) {
+	if !common_lib.ResourceManagementMeasurement_Enabled {
+		return
+	}
+	if e == nil {
+		klog.Error("Nil event")
+	}
+	checkpointsPerEvent := e.GetCheckpoints()
+	if checkpointsPerEvent == nil {
+		klog.Errorf("Event (%v, Id %s, RV %v) does not have checkpoint stamped", e.GetEventType(), e.GetId(), e.GetResourceVersionInt64())
+	}
+	lastUpdatedTime := e.GetLastUpdatedTime()
+
+	agg_received_time := checkpointsPerEvent[Aggregator_Received]
```

Comment:
if only use the LatencyMetrics.Value, which is an INT, as the map key,  it will make the hashing a bit simpler. currently the LatencyMetrics, which is the type of the event ( i.e. Aggregator_Received) is a struct of INT, TIME

Reply:
> if only use the LatencyMetrics.Value, which is an INT, as the map key, it will make the hashing a bit simpler and perform better. currently the LatencyMetrics, which is the type of the event ( i.e. Aggregator_Received) is a struct of INT, TIME

Aggregator_Received is an enum that has value 0. It is used as the index of the array